src/core/Akka/Delivery/Internal/ConsumerControllerImpl.cs

```diff
@@ -386,6 +386,17 @@ long ComputeNextSeqNr()
         Receive<Retry>(_ =>
         {
             // no retries when WaitingForConfirmation, will be performed from (idle) active
+            if (!Settings.RetryConfirmation)
+                return;
+            
+            _log.Debug("Consumer received Retry while waiting for confirmation.");
+            ReceiveRetry(() =>
+            {
+                _log.Debug("Retry sending SequencedMessage [{0}].", sequencedMessage.SeqNr);
+                CurrentState.Consumer.Tell(new Delivery<T>(sequencedMessage.Message.Message!, Context.Self,
+                    sequencedMessage.ProducerId, sequencedMessage.SeqNr));
+                Become(() => WaitingForConfirmation(sequencedMessage));
```

Comment:
Do we need to do a `Become` again here? Aren't we already in this state?

Reply:
There's no harm in doing it, we might as well have the code to explicitly show which state we're transitioning to.